src/alphabetical.py

```
@@ -0,0 +1,287 @@
+def _normalize_url(url: str) -> str:
+    """Normalize a URL for comparison:
+    - lowercase scheme and netloc
+    - remove default ports (:80, :443)
+    - strip a single trailing slash from the path
+    - preserve query/fragment
+    Returns a reconstructed URL string.
+    """
+    url = str(url)
+    try:
+        p = urlparse(url)
+    except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)